pkg/apis/spinnaker/interfaces/interfaces.go

```diff
@@ -0,0 +1,377 @@
+package interfaces
+
+import (
+	"context"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+	"reflect"
+	"time"
+)
+
+const (
+	V1alpha2Version = Version("v1alpha2")
+	LatestVersion   = V1alpha2Version
+)
+const (
+	KubernetesAccountType AccountType = "Kubernetes"
+	AWSAccountType                    = "AWS"
+)
+const (
+	Read    Authorization = "READ"
+	Write                 = "WRITE"
+	Execute               = "EXECUTE"
+)
+const (
+	HalConfigSource     = ConfigSource("hal")
+	ProfileConfigSource = ConfigSource("profile")
+)
+
+var DefaultTypesFactory = &TypesFactoryImpl{
+	Factories: map[Version]LatestTypesFactory{},
+}
+
+type ConfigSource string
+type Version string
+type AccountType string
+type AccountPermissions map[Authorization][]string
+type Authorization string
+
+type LatestTypesFactory interface {
+	NewService() SpinnakerService
+	NewServiceList() SpinnakerServiceList
+	NewAccount() SpinnakerAccount
+	NewAccountList() SpinnakerAccountList
+	NewSpinConfig() SpinnakerConfig
+	NewSpinDeploymentStatus() SpinnakerDeploymentStatus
+	NewKubernetesAuth() KubernetesAuth
+	NewHashStatus() HashStatus
+	NewExposeConfigServiceOverrides() ExposeConfigServiceOverrides
+	NewSpinnakerValidation() SpinnakerValidation
+	NewValidationSetting() ValidationSetting
+	GetGroupVersion() schema.GroupVersion
+	DeepCopyLatestTypesFactory() LatestTypesFactory
+}
+
+type TypesFactory interface {
```

Comment:
Shouldn't we be using conversions instead? As in https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/ where we'd register a conversion function into the scheme and only worry about the latest?